### PR TITLE
INF-284 adding gemspec

### PR DESCRIPTION
**Pull request by ghost**
We want to implement a way to inherit the company's ruby style guide in all our projects rather than depend on a developer having the guide cloned from github.

The implementation we were thinking would be to create a gem which inherits from both the master version of ruby-style-guide and then by BulletProof's ruby style guide.

The gem would then be a dependency of every project

### CODE REVIEW

**Comment by nerdyness** [comment]
> implemented and working with PETE.
> 
> Over to Stefan Krist to review.

Could you link me, @mfsgr ?

**Comment by nerdyness** [comment]
Thanks for [that](https://github.com/bulletproofnetworks/pete/commit/32965994911cc8519acb8ca59531add5bc3488a9#diff-11a0d7906801d9dea0eccb85667ad811R3) :+1: 

LGTM.

Do we need some documentation about versioning the Gem, how to bump the version when the .rubocop.yml file changed and that you then have to run `bundle update ruby-style-guide` etc?

**Comment by mfsgr** [comment]
@nerdyness - you reckon a comment in CONTRIBUTE.md would suffice?

**Comment by nerdyness** [comment]
> @nerdyness - you reckon a comment in CONTRIBUTE.md would suffice?

Absolutely :+1:

Maybe, apart from how to alter this Gem, also include how to include the configuration in it in a project (i.e. what you showed me yesterday, @mfsgr), an example `Gemfile` & `.rubocop.yml`.

**Comment by mfsgr** [comment]
@nerdyness - Blurb has been blurbed. Let me know what you think.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,4 +1,7 @@
+# Contributing
+
 * [Fork](https://help.github.com/articles/fork-a-repo) the project on GitHub.
 * Make your feature addition or bug fix in a feature branch. (Include a description of your changes)
 * Push your feature branch to GitHub.
-* Send a [Pull Request](https://help.github.com/articles/using-pull-requests).
\ No newline at end of file
+* Send a [Pull Request](https://help.github.com/articles/using-pull-requests).
+* Increment the ruby-style-guide version number (i.e. line 3 in **ruby-style-guide.gemspec**)
```
Comment (nerdyness) [inline CONTRIBUTING.md]: 2 quick things here:
1. Could you make the gemspec ```[a relative link](ruby-style-guide.gemspec)``` (using this syntax)?
2. What if we add a line to the top of the gemspec? We'd have to update the reference to line 3 ```Increment the ruby-style-guide version number (i.e. the "spec.version" line in [ruby-style-guide.gemspec](ruby-style-guide.gemspec))```

**Comment by nerdyness** [comment]
I like it, @mfsgr :+1:

In fact, I was showing this off to John this morning who thought it was pretty cool too :smile:

Once the doco is updated it is time to collect the juju and blog about it! hehe. We should also ping @gergnz and the T&S Council so he can add his 2cents worth. Did you want to blog about it? Not only to collect the juju but to get others to use this awesome way of distributing our cops as well.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
+# Contributing
+
 * [Fork](https://help.github.com/articles/fork-a-repo) the project on GitHub.
 * Make your feature addition or bug fix in a feature branch. (Include a description of your changes)
 * Push your feature branch to GitHub.
-* Send a [Pull Request](https://help.github.com/articles/using-pull-requests).
\ No newline at end of file
+* Send a [Pull Request](https://help.github.com/articles/using-pull-requests).
+* Increment the ruby-style-guide version number (i.e. update [**spec.version** in *ruby-style-guide.gemspec*](./ruby-style-guide.gemspec))
```
Comment (mfsgr) [inline CONTRIBUTING.md]: @nerdyness - like thus? :stuck_out_tongue_winking_eye:

**Comment by mfsgr** [comment]
@nerdyness - RE blogging, yep sounds like a good case for jujujification. Will add a task to tackle this post sprintage (i really want to get pete and billing automation out of the way)

@gergnz - As high sceptre of standards and suchness what maketh you of our gemnifications?

**Comment by gergnz** [comment]
Sounds great. We need to start building on templatising projects/products both inwards facing and customer facing. Nice work :+1:

**Pull request merged**